test/contracts/option-pair-factory.test.ts

```diff
@@ -58,7 +65,25 @@ describe('OptionPairFactory.sol', () => {
     await expect(result).to.be.revertedWith(ErrorCode.ERR_NOT_SUPPORTED);
   });
 
+  const setTreasury = async () => {
+    treasury = await deployMockContract(alice, TreasuryArtifact.abi);
+    await optionFactory.setTreasuryFor(collateral.address, treasury.address);
+  };
+
+  it('should fail to create an option pair without a treasury', async () => {
+    await collateral.mock.decimals.returns(decimals);
+    const result = optionFactory.createPair(
+      expiryTime,
+      windowSize,
+      strikePrice,
+      collateral.address,
+      referee.address
+    );
+    await expect(result).to.be.revertedWith(ErrorCode.ERR_NOT_SUPPORTED);
```

Comment:
`ERR_NOT_SUPPORTED` will be thrown first - `ERR_NO_TREASURY` will only be thrown if there's a token marked as enabled but doesn't have an existing treasury; I'm not sure whether this can happen